bigchaindb/backend/schema.py

```diff
@@ -99,3 +106,43 @@ def init_database(connection=None, dbname=None):
     create_database(connection, dbname)
     create_tables(connection, dbname)
     create_indexes(connection, dbname)
+
+
+def validate_if_exists_language(obj, key):
+    """Validate all nested "language" key in `obj`.
+
+       Args:
+           obj (dict): dictonary whose "language" key is to be validated.
+
+       Returns:
+           None: validation successfull
+
+        Raises:
+            ValidationError: raises execption incase language is not valid.
+    """
+    backend = bigchaindb.config['database']['backend']
+
+    if backend == 'mongodb':
+        data = obj.get(key, {}) or {}
+        validate_all_value_for_key(data, 'language', validate_language)
+
+
+def validate_language(value):
+    """Check if `value` is a valid language
+       https://docs.mongodb.com/manual/reference/text-search-languages/
+
+        Args:
+            value (str): language to validated
+
+        Returns:
+            None: validation successfull
+
+        Raises:
+            ValidationError: raises execption incase language is not valid.
+    """
+    if value not in VALID_LANGUAGES:
+        error_str = ('MongoDB does not support text search for the '
+                     'language "{}". If you do not understand this error '
+                     'message then please rename key/field "language" to '
+                     'something else like "lang".').format(value)
+        raise ValidationError(error_str) from ValueError()
```

Comment:
the `raise ... from` idiom is used to raise a new exception keeping the context of another exception that already happened. I'm not sure the usage here is correct, I guess the correct line should be just a simple `raise ValidationError(error_str)`.